org.eclipse.lsp4e/src/org/eclipse/lsp4e/LanguageServers.java

```diff
@@ -360,6 +384,21 @@ private <T> void completeEmptyOrWithException(final CompletableFuture<Optional<T
 		}
 	}
 
+	/**
+	 * Gets the modification stamp for the supplied document, or returns -1 if not available.
+	 *
+	 * In practice just a sanity-checked downcast of a legacy API: should expect the platfom to be instantiating
+	 * Documents that implement the later interfaces.
+	 *
+	 * Should be called on UI thread
+	 *
+	 * @param document Document to check
+	 * @return Opaque version stamp, or -1 if not available
+	 */
+	public static long getDocumentModificationStamp(@Nullable IDocument document) {
```

Comment:
should this not be private?

Reply:
Package-private now (`VersionedEdits.apply()` needs it)